File: leo2-common/src/main/java/org/deku/leo2/messaging/MessagingContext.java
package org.deku.leo2.messaging;

import sx.jms.embedded.Broker;

import javax.jms.Queue;
import javax.jms.Topic;

/**
 * Generic messaging context interface definition
 * Created by masc on 07.06.15.
 */
public interface MessagingContext {
    /** Broker for this messaging context */
    Broker getBroker();

    /** Central message queue for lightweight message exchange */
    Queue getCentralQueue();
    /** Central message queue for entity sync */
    Queue getCentralEntitySyncQueue();
    /** Central message queue for logs */
    Queue  getCentralLogQueue();
    /** Node message queue for lightweight message exchange*/
    Queue  getNodeQueue(Integer id);
    /** Node notification topic for broadcastss */
    Topic getNodeNotificationTopic();
}

File: libs/leoz-common/src/main/java/org/deku/leoz/messaging/MessagingContext.kt
package org.deku.leoz.messaging

import sx.jms.embedded.Broker

import javax.jms.Queue
import javax.jms.Topic

/**
 * Common messaging context interface definition
 * Created by masc on 07.06.15.
 */
interface MessagingContext {
    /** Broker for this messaging context  */
    val broker: Broker

    /** Central message queue for lightweight message exchange  */
    val centralQueue: Queue
    /** Central message queue for entity sync  */
    val centralEntitySyncQueue: Queue
    /** Central message queue for logs  */
    val centralLogQueue: Queue

    /** Node message queue for lightweight message exchange */
    fun getNodeQueue(id: Int): Queue

    /** Node notification topic for broadcastss  */
    val nodeNotificationTopic: Topic
}
